Automattic/jetpack
# Move contact form PHP files

Part of #28405.

## Proposed changes:


This patch moves contact form PHP files from `plugins/jetpack/modules/contact-form` into `packages/forms/src/contact-form`.

I used this as an opportunity to organize the classes into separate files and move everything into object-oriented structures. There's still a lot of gradual improvements to be done but I think that's already a big step forward.  

**Note:** This PR will require #28417 to be merged, and the static file references to be updated before it can be fully tested.

### Other information:


## Jetpack product discussion



pcsBup-N3-p2

## Does this pull request change what data or activity we track or use?



No.

## Testing instructions:






Use `composer phpunit` inside the package directory to run unit tests.

Enable the package on your jetpack install by using the `jetpack_contact_form_use_package` filter:
```
add_filter(
    'jetpack_contact_form_use_package',
    function () {
        return true;
    }
);
```

- Contact form module should load as expected.
- Verify that you can load the feedback area in the dashboard and the filters work as expected.
- Verify that you can export feedback as expected.
- Verify that the contact form gutenberg blocks load as expected in the editor.
- Verify that the contact form blocks render as expected on published pages or posts.
